tests/unit/test_base_journey.py

```diff
@@ -122,3 +122,24 @@ def test_verify_thresholds_collection_shape_invalid_shape(caplog):
         )
     assert not is_valid
     assert "mapping" in caplog.text
+
+
+@pytest.mark.parametrize(
+    "input_url,expected_output",
+    [
+        ("https://test_url.com/", "https://test_url.com"),
+        ("  https://test_url.com/path  ", "https://test_url.com"),
+        ("http://test_url.com/some/path/", "http://test_url.com"),
+        ("https://test_url.com//", "https://test_url.com"),
```

Comment:
Do we expect all these as valid inputs to normalize_url? Should we be removing /some/path from the input for example?

If that is true, then maybe call the function as parse_base_url instead of normalize_url.